music_grabber.py

```diff
@@ -23,6 +20,7 @@ class WrongFileSizeError(ValueError):
 class MusicDownloader:
 
     def download(self, input_url, filesize_limit):
+        input_url = input_url.split('&list', 1)[0]
```

Comment:
why did you have to add this?

Reply:
that's for youtube videos link which contains playlist id. when such link passed to youtub-dl, youtube-dl will download the whole playlist.
`/sets/` and `?in=` (for soundcloud) in text_handler is also used for filter playlists out.

```
# youtube video in playlist, all items in the playlist will be downloaded
https://www.youtube.com/watch?v=7PBYGu4Az8s&list=PLFgquLnL59alCl_2TQvOiD5Vgm1hCaGSI&index=1
# soundcloud playlist
https://soundcloud.com/jasonlaville/sets/best-deep-house-chill-music
# soundcloud track in playlist, but only one song will be downloaded
https://soundcloud.com/deependmusic/matt-simons-catch-release-deepend-remix-out-now?in=jasonlaville/sets/best-deep-house-chill-music`

Reply:
ah i see, but if you pass a link that does not contain a **list id** like
`https://www.youtube.com/watch?v=7PBYGu4Az8s`
then input_url.split('&list', 1)[0] will just return
`https://www.youtube.com/watch?v=7PBYGu4Az8s`
?

Reply:
also why didn't you do the same for `?in=`
something like `input_url.split('?in=', 1)[0]`
instead you decided not to download songs that have `?in=`

Reply:
> but if you pass a link that does not contain a list id like
> https://www.youtube.com/watch?v=7PBYGu4Az8s
> then
> input_url.split('&list', 1)[0]
> will just return
> https://www.youtube.com/watch?v=7PBYGu4Az8s

Yes, already tested before commit that.

> also why didn't you do the same for ?in=
> something like input_url.split('?in=', 1)[0]
> instead you decided not to download songs that have ?in=

because for youtube video with `&list` it will download the playlist, but for soundcloud, it won't but download the track instead.

idk if it's because of the difference between the webpage design of youtube and soundcloud. anyway we don't need that for now. we can easily change that when there are changes in the future or we have any problem with that.